compat/v0/compat.go

```diff
@@ -687,20 +724,23 @@ func V1TxFromV0Burn(ctx context.Context, v0tx Tx, bindings *binding.Binding, net
 
 func V0TxHashFromTx(tx Tx) (B32, error) {
 	var hash B32
+	// Incorrect - we need to use the nhash, not the nonce
 	if IsShiftIn(tx.To) {
-		payload := pack.NewBytes(tx.In.Get("p").Value.(ExtEthCompatPayload).Value[:])
-		phash := engine.Phash(payload)
-		tokenAddr := tx.In.Get("token").Value.(ExtEthCompatAddress)
-		to := tx.In.Get("to").Value.(ExtEthCompatAddress)
-		n := tx.In.Get("n").Value.(B32)
-		var nonce pack.Bytes32
-		copy(nonce[:], n[:])
-		ghash, err := engine.V0Ghash(tokenAddr[:], phash, to[:], nonce)
-		if err != nil {
-			return B32{}, err
-		}
-		utxo := tx.In.Get("utxo").Value.(ExtBtcCompatUTXO)
-		copy(hash[:], crypto.Keccak256([]byte(fmt.Sprintf("txHash_%s_%s_%s_%d", tx.To, ghash, utxo.TxHash, utxo.VOut.Int.Int64()))))
+		return hash, fmt.Errorf("cannot handle shift-ins")
```

Comment:
Maybe rather than returning an error here, we should panic and rename the function to `V0TxHashFromBurn`.